test/test_jit.py

```diff
@@ -3947,6 +3947,53 @@ def test_append():
             return a == [0, 1, 2, 3]
         self.checkScript(test_append, ())
 
+    def test_comprehensions_basic(self):
+
+        @torch.jit.script
+        def comp(l):
+            # type: (List[int]) -> List[int]
+
+            n = [x * 3 for x in l]
+            return n
+
+        self.assertEqual(comp([1, 2, 3]), [3, 6, 9])
+
+    def test_comprehensions_basic_float(self):
+
+        @torch.jit.script
+        def comp(l):
+            # type: (List[float]) -> List[float]
+
+            n = [x * 3 for x in l]
+            return n
+
+        self.assertEqual(comp([1.0, 2.0, 3.0]), [3.0, 6.0, 9.0])
+
+    def test_comprehensions_two_comps(self):
+
+        @torch.jit.script
+        def comp(l1, l2):
+            # type: (List[int], List[int]) -> List[int]
+
+            n = [x * 3 for x in l1]
+            n2 = [x + 2 for x in l2]
+            return n + n2
+
+        self.assertEqual(comp([1, 2, 3], [4, 5]), [3, 6, 9, 6, 7])
+
+    def test_comprehensions_wrong_expr_type(self):
+
```

Comment:
formatting nit: generally we don't leave blank lines after `def`, `with` or type annotations